DirectEmployers/MyJobs
# [PD-1866] Wrapped PRM views with requires on QC and Staging

# Description
I've modified the `requires`, and `warn_when` family of decoraties (including `has_access`) decorators as follows:
- `requires` returns the original view if `settings.DEBUG` is `False`, otherwise it does its thug thizzle.
- `warn_when` is unconditionally enabled unless the `enabled` argument is set to `False`. In which case, it is disabled when `settings.DEBUG` is `True`.

The effect of the above is that if a view is decorated with both `has_access` and `requires`, requires will only work when debugging is enabled, where as `has_access` will only work when debugging is disabled.

# Manually Testing
In order to test this yourself, modify the `DEBUG` variable in your `settings.py` and observe the following expected behavior:
- When `DEBUG` is `True`:
-- visiting any PRM page for a company that doesn't have "PRM" app access should result in a 404 page saying that app-level permissions are missing
-- visiting any PRM page for a company that *does* have "PRM" app access, but as a user whose roles don't have all of the PRM-related activities should result in a 404 saying that activities are missing.
- When `DEBUG` is `False`:
-- app access, roles, and activities shouldn't matter. Instead, you should see a warning when attempting to access PRM for a company who is not a member. 

# Caveats
- I am aware that it is possible to assign a user to a role that belongs to a company for which they do not have access. As roles will eventually accomplish what having a row on the company user table does, becoming a so-called company user would be a matter of being assigned a role in that company with the appropriate activities. In the mean time, I think we should do any role filtering at the UX level.

# Relevant Changes
I based this off of my pd-1850 branch. If you're curious about what is different between that one and this one, go [here](https://github.com/DirectEmployers/MyJobs/compare/pd-1850...pd-1866). Should be less intimidating than what's currently showing in the "Files changed" tab.
